src/launch/java/baritone/launch/mixins/MixinScreen.java

```diff
@@ -36,15 +33,19 @@
 public abstract class MixinScreen implements IGuiScreen {
 
     //TODO: switch to enum extention with mixin 9.0 or whenever Mumfrey gets around to it
-    @Inject(at = @At(value = "INVOKE", target = "Lorg/slf4j/Logger;error(Ljava/lang/String;Ljava/lang/Object;)V", remap = false, ordinal = 1), method = "handleComponentClicked", cancellable = true)
-    public void handleCustomClickEvent(Style style, CallbackInfoReturnable<Boolean> cir) {
+    @Inject(method = "handleComponentClicked", at = @At(
+        value = "INVOKE",
+        target = "Lnet/minecraft/client/multiplayer/ClientPacketListener;sendUnsignedCommand(Ljava/lang/String;)Z"
+    ), cancellable = true)
+    public void handleCustomClickEvent(final Style style, final CallbackInfoReturnable<Boolean> cir) {
         ClickEvent clickEvent = style.getClickEvent();
         if (clickEvent == null) {
             return;
         }
+        if (!(clickEvent instanceof ClickEvent.RunCommand(String command))) return;
```

Comment:
Cursed, but kinda nice?
Didn't know we can pattern match now. (only knew about `o instanceof C o_`)

Reply:
Quite sure this breaks if a book for example contains a click event running a command (as you are cancelling the original code away)

Reply:
You are right, this cancels any commands, even if Baritone doesn't handle them.

Now  I do wonder though: Didn't the previous injection do the same thing?

Reply:
No because the injection point was different, the new injection point is also wrong I think. The code was previous only called when the command sending failed I think.

Reply:
Ok, fell for a little gotcha (tried to use `/say hi` in a click event) and the command generator I used was broken, but I managed to confirm that Baritone does correctly handle clickable text in 1.19.4 and does not allow running vanilla commands in 1.21.5.

Minecraft (1.19.4) normally displays an error message when running a chat command without a leading slash, so my guess is that the previous injection point was exactly that error message, meaning it would only run for Baritone commands (or from anything else using anything but "/" as its prefix).
Problem with that behavior is that Minecraft (1.21.5) allows commands without prefix in click events now.

Reply:
> the new injection point is also wrong I think

Because it is triggered for vanilla commands as well or is there another reason?

Reply:
> > the new injection point is also wrong I think
> 
> Because it is triggered for vanilla commands as well or is there another reason?

This was where the old injection was placed, so the code was never called when the command sending succeeded.
![image](https://github.com/user-attachments/assets/825ca269-7f2e-40d4-bcdf-b13f89c7285c)

Reply:
Maybe both the highlighted line and the second `LOGGER.error` call were targeted? Otherwise e.g. the text from Baritones help menu was broken.

I have a potential fix, I'm just trying to fit it into GitHubs incredibly limited "suggest a diff" feature.

Reply:
Maybe both the highlighted line and the second `LOGGER.error` call were targeted? Otherwise e.g. the text from Baritones help menu cannot work (it starts commands with "<<", definitely not with "/").

Reply:
Nah the ordinal was set to 1, meaning only the highlighted one was targetted. They changed this behaviour already in the past (in 1.19 versions) so I could imagine the Mixin silently broke without anyone noticing. (For the exact behaviour changes, see https://github.com/ViaVersion/ViaFabricPlus/blob/main/src/main/java/com/viaversion/viafabricplus/injection/mixin/features/run_command_action/MixinScreen.java#L46)

Reply:
Here's what I think should make this work. Not usable as is because it needs an additional import statement.
```suggestion
        if (!(clickEvent instanceof ClickEvent.RunCommand(String command))) return;
        if (!command.startsWith(FORCE_COMMAND_PREFIX)) {
            return;
        }
```

Reply:
Isn't `ordinal` zero-based and the two calls in the screenshot the only candidates, so the second one is targeted?

Reply:
There is another logger usage above which is not part of the screenshot

Reply:
That one (and the fourth one below) has the wrong signature.